public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -651,7 +651,7 @@ function errorAndDurationQuery(
 
   let labels = [];
   if (request.app === CoreApp.Explore) {
-    if (rateResponse.data[0][0]?.fields[1]?.values) {
+    if (rateResponse.data[0][0]?.fields[1]?.name === 'span_name' && rateResponse.data[0][0]?.fields[1]?.values) {
```

Comment:
Can we rely on the order of the fields here?

Reply:
Unless Prometheus changes the order of fields in the response or something else unknown causes a reorder. No need for us to be reliant on this ordering though, so updated the logic to find the field instead.